STYLEGUIDE.md

```diff
@@ -0,0 +1,716 @@
+
+❌ Examples of **incorrect** code:
+```ts
+/* Don't use for ... of if you can use .forEach */
+for (const item of array) {
+  ...
+}
+
+/* Don't use .forEach if you can use .map */
+const result = [];
+array.forEach(item => {
+  result.push(item.property);
+});
+```
+
+✅ Examples of **correct** code:
+```ts
+/* Use .forEach if possible */
+array.forEach(item => {
+  ...
+});
+
+/* Use .map, if appropriate */
+const result = array.map(item => item.property);
+```
+
+
+## `null` and `undefined`
+
+### Use optional parameters for optional values
+Use optional parameters, if there is a default value or if it semantically makes sense to omit the parameters when consuming the API.
+
+❌ Examples of **incorrect** code:
+```ts
+/* Don't use explicit undefined if the parameter is semantically optional */
+function foo(options: Options | undefined) {
+  ...
+}
+```
+
+✅ Examples of **correct** code:
+```ts
+/* Use optional parameter if the parameter is semantically optional */
+function foo(options?: Options) {
+  ...
+}
+
+/* Use default parameter if there is a default (making the parameter inherently optional for the consumer) */
+function foo(options: Options = {}) {
+  ...
+}
+```
+
+### Use `undefined` for explicit absence of values
+If a value semantically is not optional prefer using `undefined` for values that do not exist or are not defined.
+
+❌ Examples of **incorrect** code:
+```ts
+/* Don't use null for non-existent values */
+function foo(obj: SomeType | null) {
+  ...
+}
+```
+
+✅ Examples of **correct** code:
+```ts
+/* Use undefined for non-existent values */
+function foo(obj: SomeType | undefined) {
+  ...
+}
+```
+
+### Use `null` for intentional explicit absence of values or if it is part of an external API
+If a value can both be non-existent and intentionally absent and those states are semantically different it may make sense to allow the usage of `null`.
+Those cases should however be rather rare and are more common when calling external APIs, e. g. through HTTP requests.
+Consider whether there is a better API design that might avoid this.
+
+❌ Examples of **incorrect** code:
+```ts
+/* Don't use null for absence in general */
+function foo(obj?: SomeType | null) {
+  if (obj === null || obj === undefined) {
+    ...
+  }
+}
+```
+
+✅ Examples of **correct** code:
+```ts
+/* Use null if it semantically makes sense and is different from undefined */
+function foo(obj: SomeType | null | undefined) {
+  if (obj === null) {
+    ...
+  } else if (obj === undefined) {
+    ...
+  }
+}
+```
+
+### Use truthy/falsy checks where possible
+In most cases it is possible to check for truthyness/falsyness instead of explicitly comparing values.
+This should be used when possible, but carefully considered in cases where falsy values are valid and therefore semantically truthy values, e. g. 0, ''.
+Of course more fine granular checks should be applied if semantically needed.
```

Comment:
```suggestion
### Use truthy/falsy checks where possible
In most cases, it is possible to check for truthiness/falseness instead of explicitly comparing values.
This should be used when possible, but carefully considered in cases where falsy values are valid and therefore semantically truthy values, e. g. 0, ''.
Of course, more fine granular checks should be applied if semantically needed.
```

Reply:
Oh good catch, but it still should be falsiness - I think there are a few more occurences, I will fix them with replace all.